pkg/component/component.go

```diff
@@ -66,7 +66,7 @@ func CreateFromGit(client *occlient.Client, name string, ctype string, url strin
 	// save source path as annotation
 	annotations := map[string]string{componentSourceURLAnnotation: url}
 
-	output, err := client.NewAppS2I(name, ctype, url, labels, annotations)
+	output, err := client.NewAppS2I(currentApplication, name, ctype, url, labels, annotations)
```

Comment:
`name` argument in `NewAppS2I` is not a name of the component. It is name for OpenShift objects, we have been just passing the same value as component name till now.

You can pass in new namespaced name here without the need to add a new argument to that function.

The component name is determined by label (`GetLabels`)

Reply:
I'll refactor this :+1: thx!